numpy/ma/core.py

```diff
@@ -3121,6 +3121,13 @@ def __getitem__(self, indx):
             if isinstance(indx, basestring):
                 if self._fill_value is not None:
                     dout._fill_value = self._fill_value[indx]
+                    # If we're indexing a multidimensional field in a
+                    # structured array (such as dtype("(2)i2,(2)i1")),
+                    # indexing fill_value will yield a multi-element
```

Comment:
@seberg Is that something you'd want to check runtime every time it is accessed?  I haven't tested it, but mightn't that cause some serious overhead?